### ISSUE TITLE
[Accessibility, DAP Violation] ariaLabel property unused on Carbon Dropdown for underlying button

### ISSUE THREAD

**Issue by a-roberts** [carbon-design-system/carbon#3997]
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

Specifically (ish), carbon-components": "^10.6.1" and "carbon-components-react": "^7.6.1",

## Detailed description

For reference see https://github.com/tektoncd/dashboard/issues/524 - looks like the ariaLabel property isn't being picked up when added to the SideNavMenuItem component, e.g. with

```
          <SideNavMenuItem
            element={NamespacesDropdown}
            id="sidenav-namespace-dropdown"
            selectedItem={{
              id: namespace,
              text: namespace,
              ariaLabel: namespace
            }}
            showAllNamespaces
            onChange={this.selectNamespace}
          >
```

## Is this issue related to a specific component?`
Yeah, SideBarNavItem I think

## What did you expect to happen? What happened instead? What would you like to see changed?

ariaLabel should be passed to the underlying button

## What browser are you working in?

Any browser, but I'm using Chrome with the inspector up

## What version of the Carbon Design System are you using?

How do I check this?

## What offering/product do you work on? Any pressing ship or release dates we should be aware of?

Tekton Dashboard and Webhooks Extension, likely have a workaround

## Steps to reproduce the issue

- Screenshots or code

![image](https://user-images.githubusercontent.com/12814972/64970078-06f9a580-d89d-11e9-942f-ea2f9e4bafe9.png)

Sandbox code (can't figure out how to share, share button just links me to a new editor, forking never returns):

```
import { render } from "react-dom";
import { Dropdown } from "carbon-components-react";
import React from "react";

import {
  SideNav as CarbonSideNav,
  SideNavItems,
  SideNavMenu,
  SideNavMenuItem
} from "carbon-components-react";

const NamespacesDropdown = props => {
  return <Dropdown {...props} />;
};

const allNamespacesLabel = "All Namespaces";

NamespacesDropdown.defaultProps = {
  allNamespacesLabel,
  items: [],
  loading: false,
  label: "Select Namespace",
  ariaLabel: "Select Namespace",
  titleText: "Namespace",
  emptyText: "No Namespaces found",
  showAllNamespaces: false
};

const App = () => (
  <div>
    <CarbonSideNav
      isFixedNav
      expanded
      isChildOfHeader={false}
      aria-label="Side navigation"
    >
      <SideNavItems>
        <SideNavMenu defaultExpanded title="Tekton">
          <SideNavMenuItem element={NamespacesDropdown} />
        </SideNavMenu>
      </SideNavItems>
    </CarbonSideNav>
  </div>
);

render(<App />, document.getElementById("root"));
```

I expect "Select Namespace" to be used as the aria label on the button instead of Open Menu

Also you can see this is case for the Dropdown component itself just using

http://react.carbondesignsystem.com/?path=/story/dropdown--default

![image](https://user-images.githubusercontent.com/12814972/65025654-06a9ea80-d92f-11e9-890d-9d87d298a138.png)

**Comment by dakahn** [comment]
@asudoh any ideas? I think this is related to their implementation and maybe not the underlying component?

**Comment by asudoh** [comment]
Given we don't have the reduced case, it's hard for us to see what the problem is. My recommendation for the issue author is either debugging the corresponding file/line in `node_modules/carbon-components-react/es` or `node_modules/carbon-components-react/lib` to https://github.com/carbon-design-system/carbon/blob/v10.6.0/packages/react/src/components/Dropdown/Dropdown.js#L271 or providing a reduced case.

**Comment by a-roberts** [comment]
@asudoh is the sandbox link I've given not a reduced case? 

Including @alangreene as an FYI (we work together on the Tekton Dashboard which is where we're using the component), thanks

**Comment by asudoh** [comment]
I only see below in this issue - Probably you want to re-post the link?

## Steps to reproduce the issue

1. todo

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

todo

**Comment by a-roberts** [comment]
@asudoh 😸 it's under additional info

**Comment by asudoh** [comment]
For that, it may yield more help if you can turn that to a live example based on https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

**Comment by AlanGreene** [comment]
@asudoh the issue can be reproduced using the Carbon React storybook as Adam mentioned.

http://react.carbondesignsystem.com/?path=/story/dropdown--default

**Comment by asudoh** [comment]
Without modification? I was expecting something like going to a site and being able to reproduce the problem.

**Comment by abbeyhrt** [comment]
@a-roberts Could you share a screenshot of the DAP violation?

**Comment by a-roberts** [comment]
@abbeyhrt everything's now fine for our dropdown actually, I looked in the node_modules folder and observed carbon components 10.6.1 and 7.6.1 of carbon-components-react. 

Had a look through our source code (https://github.com/tektoncd/dashboard, it was the namespaces dropdown that was impacted) and don't see any relevant changes, unless @AlanGreene knows of anything else that could have caused this.

Thanks if a fix was provided 😄

**Comment by abbeyhrt** [comment]
@a-roberts okay great! I'm going to close this for now but feel free to open it back up if the problem returns!
